### PR TITLE
Navtabs

### PR DESCRIPTION
**Pull request by ghost**
yoo ricky finished up tabs/active signal/templates here. @project-signal

### CODE REVIEW

File: app/assets/javascripts/components/dashboard/navigation/content_panel/content_panel.jsx [hunk 1]
```diff
@@ -1,8 +1,46 @@
...
+    const responses = (signal) => {
+      const data = [];
+
+      _.forEach(signal.responses, function(key) {
+        data.push(state.models.responses.data[key]);
+      });
+      return data;
+    };
+
+    const newSignal = (signal, responses) => {
```
Comment (ricardo-quinones) [inline app/assets/javascripts/components/dashboard/navigation/content_panel/content_panel.jsx]: You can use lodash's `each` method here for succinctness.

File: app/assets/javascripts/components/dashboard/navigation/content_panel/content_panel.jsx [hunk 2]
```diff
@@ -1,8 +1,46 @@
...
+      const newSignalObj = {};
+
+      Object.keys(signal).forEach(function(key) {
```
Comment (ricardo-quinones) [inline app/assets/javascripts/components/dashboard/navigation/content_panel/content_panel.jsx]: Sorry, I meant here.

File: app/assets/javascripts/components/dashboard/navigation/content_panel/content_panel.jsx (hunk 1 above)
Comment (ricardo-quinones) [reply app/assets/javascripts/components/dashboard/navigation/content_panel/content_panel.jsx]: Let's not define the function withing this function. We can define it outisde of it.

File: app/assets/javascripts/components/dashboard/navigation/content_panel/content_panel.jsx [hunk 3]
```diff
@@ -1,8 +1,46 @@
...
+  if (route.path === 'new') {
+    return {
+      type: params.type,
+    }
+  } else {
+    const signals = state.models.listenSignals.data;
+    const key = _.findKey(signals, { id: parseInt(params.id) });
+    const signal = signals[key];
+
+    const responses = (signal) => {
```
Comment (ricardo-quinones) [inline app/assets/javascripts/components/dashboard/navigation/content_panel/content_panel.jsx]: Define the function outisde of the function.

**Review by ricardo-quinones — COMMENTED** [review]
This looks great! One thing you should think about is what happens when someone types in a random signal id URL in the address bar. If it was just Rails routing, you would 404. But how are we going to handle it on the client-side?

One thing that's nice about redial and how we're using it with react router is that we can fail routes on a hook. So, we could add a hook that ensures that the signal id is valid before routing there. Something to think about.

File: app/assets/javascripts/components/dashboard/navigation/content_panel/content_panel.jsx (hunk 3 above)
Comment (ghost) [reply app/assets/javascripts/components/dashboard/navigation/content_panel/content_panel.jsx]: Ahh yea I noticed it when i was just refreshing and realized the the props/states are only passed down when it's in the happy path.. I'll take care of that

**Comment by ghost** [comment]
@ricardo-quinones let me merge the v1 braaaaaa

**Comment by ricardo-quinones** [comment]
Go head brah

**Pull request merged**